### PR TITLE
Aborts heartbeat if the configured auth_key is invalid

### PR DESCRIPTION
**Pull request by OfficialLambdax**
checked 17 or so keys for their format

### CODE REVIEW

**Comment by Lucky4Luuk** [comment]
Overall looks good but would it not be better to simply pass the auth key to the backend and stop sending heartbeats if the backend informs us the key is invalid? If the backend ever changes format, the server wouldn't need an update in that case, because it just asks the backend if it's valid or not.

**Comment by OfficialLambdax** [comment]
Theoretically i would agree but the backend downtimes can also lead to the abort of heartbeating. So imagine running a 24/7 server and in one of the beats the backend fails.. now the server will go in private until the owner notices. thats why im against doing this based on the backend's response

**Comment by Lucky4Luuk** [comment]
That's a valid point, that could be a problem. I'll merge this change for now, if the format of the auth key ever gets changed, we'll just have another look at it :)

File: src/config.rs [hunk 1]
```diff
@@ -43,3 +43,21 @@ pub struct GeneralSettings {
     #[serde(rename = "ResourceFolder")]
     pub resource_folder: String,
 }
+
+impl GeneralSettings {
+    pub fn is_auth_key_valid(&self) -> bool {
```
Comment (lionkor) [inline src/config.rs:48]: https://crates.io/crates/uuid

Comment (Lucky4Luuk) [reply src/config.rs:48]: yes that does make sense

**Pull request merged**